### PR TITLE
Translate [c++/en] into [c++/cn]

### PR DESCRIPTION
**Pull request by Arnie97**

### CODE REVIEW

File: zh-cn/c++-cn.html.markdown [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+    // 父类中的函数必须被显式声明为_虚函数（virtual）_。
+    // 考虑到性能方面的因素，函数默认情况下不会被声明为虚函数。
+    virtual void print() const;
+
+    // 函数也可以在class body内部定义。
+    // 这样定义的函数会自动成为内联函数。
+    void bark() const { std::cout << name << " barks!\n" }
+
+    // 除了构造器以外，C++还提供了析构器。
+    // 当一个对象被删除或者脱离其定义域时时，它的析构函数会被调用。
```
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: Typo

File: zh-cn/c++-cn.html.markdown [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+    // 这使得RAII这样的强大范式（参见下文）成为可能。
+    // 为了衍生出子类来，基类的析构函数必须定义为虚函数。
+    virtual ~Dog();
+
+}; // 在类的定义之后，要加一个分号
```
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: Duplicate line

File: zh-cn/c++-cn.html.markdown [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+{
+    cout << "Goodbye " << name << "\n";
+}
+
+void Dog::setWeight(int dogsWeight)
+{
+    weight = dogsWeight;
+}
+
+// 虚函数的virtual关键字只需要在声明时使用，不需要在定义时重复
```
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: Duplicated function here.

File: zh-cn/c++-cn.html.markdown [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+// 继承：
+
+// 这个类继承了Dog类中的公有（public）和保护（protected）对象
+class OwnedDog : public Dog {
+
+    void setOwner(const std::string& dogsOwner)
+
+    // 重写OwnedDogs类的print方法。
+    // 如果你不熟悉子类多态的话，可以参考这个页面中的概述：
+    // http://en.wikipedia.org/wiki/Polymorphism_(computer_science)#Subtyping
```
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: Is there a Chinese page for polymorphism we can use here?

File: zh-cn/c++-cn.html.markdown [hunk 5]
```diff
@@ -0,0 +1,590 @@
...
+    if (fh == nullptr)
+        throw std::exception("Could not open the file.");
+
+    try {
+        doSomethingWithTheFile(fh);
+        doSomethingElseWithIt(fh);
+    }
+    catch (...) {
+        fclose(fh); // 保证出错的时候文件被正确关闭
+        throw; // Then re-throw the exception.
```
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: Missed a line here.

File: zh-cn/c++-cn.html.markdown [hunk 6]
```diff
@@ -0,0 +1,590 @@
...
+    doSomethingWithTheFile(fh);
+    doSomethingElseWithIt(fh);
+
+} // 文件已经被析构器自动关闭
+
+// 与上面几种方式相比，这种方式有着_明显_的优势：
+// 1. 无论发生了什么情况，资源（此例当中是文件句柄）都会被正确关闭。
+//    只要你正确使用了析构器，就_不会_因为忘记关闭句柄，造成资源的泄漏。
+// 2. 可以注意到，通过这种方式写出来的代码十分简洁。
+//    析构器会在后臺关闭文件句柄，不再需要你来操心这些琐事。
```
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: How did that traditional character get in there? :)

File: zh-cn/c++-cn.html.markdown (hunk 5 above)
Comment (Arnie97) [inline zh-cn/c++-cn.html.markdown]: Thanks.

Comment (Arnie97) [inline zh-cn/c++-cn.html.markdown]: @geoffliu Thanks.

File: zh-cn/c++-cn.html.markdown (hunk 1 above)
Comment (Arnie97) [inline zh-cn/c++-cn.html.markdown]: @geoffliu Should I just add spaces around English words?

Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: Oh I just mean you have two "时"

File: zh-cn/c++-cn.html.markdown (hunk 4 above)
Comment (geoffliu) [inline zh-cn/c++-cn.html.markdown]: http://zh.wikipedia.org/wiki/%E5%AD%90%E7%B1%BB%E5%9E%8B

Let's use this.

**Comment by geoffliu** [comment]
Awesome! Thanks!

**Pull request merged**